sql/internal/sqlx/migrate.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021-present The Atlas Authors. All rights reserved.
+// This source code is licensed under the Apache 2.0 license found
+// in the LICENSE file in the root directory of this source tree.
+
+package sqlx
+
+import (
+	"context"
+
+	"ariga.io/atlas/sql/migrate"
+	"ariga.io/atlas/sql/migrate/ent"
+	"ariga.io/atlas/sql/migrate/ent/revision"
+	"entgo.io/ent/dialect/sql"
+)
+
+// A RevisionStorage provides implementation for the migrate.RevisionReadWriter interface.
+type RevisionStorage struct {
+	c *ent.Client
+	// sc is the function signature of the Ent migration engine.
+	// Due to cyclic dependencies between Ent and Atlas, this value is stitched in at runtime.
+	sc func(context.Context) error
+}
+
+// NewRevisionStorage creates a new RevisionStorage with the given ent.Client.
+func NewRevisionStorage(c *ent.Client) *RevisionStorage {
+	return &RevisionStorage{c: c}
+}
```

Comment:
```suggestion
func NewRevisionStorage(dialect string, db schema.ExecQuerier) *RevisionStorage {
	return &RevisionStorage{
		c: ent.NewClient(ent.Driver(sql.NewDriver(sql.Conn{ExecQuerier: db}, dialect))),
	}
}
```

Then, in drivers, call:

```go
sqlx.NewRevisionStorage(dialect.Postgres, db)
sqlx.NewRevisionStorage(dialect.MySQL, db)
```